### PR TITLE
test: let the keyboard layout tester produce YAML as output format

### PR DESCRIPTION
**Pull request by whot**
We're compiling something like 80k layout combinations in the xkeyboard-config CI with this tool, producing a 6+ GB output file that we then have to grep through to find the actual error. That is ... not very productive.

Let's switch to this tool printing out YAML format so we can use `yq` or other scripts to search for the data we want.

An example MR for how the results then look like is here: https://gitlab.freedesktop.org/whot/xkeyboard-config/-/merge_requests/1, just click on the "Expand" button once the test results are parsed and it nicely shows which ones break.

cc @svu

### CODE REVIEW

File: test/xkeyboard-config-test.py.in [hunk 1]
```diff
@@ -209,6 +221,22 @@ def run(combos, tool, njobs):
...
+            if keymap_output_dir:
+                # we're running through the layouts in a somewhat sorted manner,
+                # so let's keep the fd open until we switch layouts
+                layout = invocation.layout
+                if invocation.variant:
+                    layout += f"({invocation.variant})"
+                fname = keymap_output_dir / layout
+                if fname != keymap_file:
+                    keymap_file = fname
+                    keymap_file_fd = open(keymap_file, 'a')
```
Comment (bluetech) [inline test/xkeyboard-config-test.py.in:233]: Probably should  close the old one first

File: test/xkeyboard-config-test.py.in [hunk 2]
```diff
@@ -265,6 +265,10 @@ def main(args):
     parser.add_argument('--verbose', '-v', default=False, action="store_true")
     parser.add_argument('--keymap-output-dir', default=None, type=str,
                         help='Directory to print compiled keymaps to')
+    parser.add_argument('--layout', default=None, type=str)
```
Comment (bluetech) [inline test/xkeyboard-config-test.py.in:268]: Maybe a help "Only test this specific layout" etc?

**Review by bluetech — APPROVED** [review]
LGTM.

I think it would be good to mention somewhere in the code and/or help that the output is YAML.

File: test/xkeyboard-config-test.py.in (hunk 1 above)
Comment (whot) [reply test/xkeyboard-config-test.py.in:233]: fixed, thanks

File: test/xkeyboard-config-test.py.in (hunk 2 above)
Comment (whot) [reply test/xkeyboard-config-test.py.in:268]: help added, thanks

**Comment by whot** [comment]
> I think it would be good to mention somewhere in the code and/or help that the output is YAML.

Added a commit to expand the `--help` output to mention that

**Pull request merged**